mautrix/bridge/user.py

```diff
@@ -162,9 +166,34 @@ async def push_bridge_state(
         if state.should_deduplicate(self._prev_bridge_status):
             return
         self._prev_bridge_status = state
-        await state.send(
-            self.bridge.config["homeserver.status_endpoint"], self.az.as_token, self.log
-        )
+        self._bridge_state_queue.append(state)
+        if not self._bridge_state_loop:
+            self.log.debug(f"Starting bridge state loop")
+            self._bridge_state_loop = asyncio.create_task(self._start_bridge_state_send_loop())
+        else:
+            self.log.debug(f"Queued bridge state to send later: {state.state_event}")
+
+    async def _start_bridge_state_send_loop(self):
+        while self._bridge_state_queue:
+            state = self._bridge_state_queue[-1]
```

Comment:
Here you are removing from the end of the list, but later you are popping from the front of the list.

Reply:
🤦

Reply:
BTW, `pop` returns the element at the index provided, so you probably just want to use that and re-add to the end of the list on failure.

Reply:
I opted not to pop and replace, just because I try to err on the side of mutating state as little as possible, makes it harder for me to break things in the future.